test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/SinceTagSniff.php

```diff
@@ -0,0 +1,614 @@
+	 * @return array An array containing violation codes for missing since tag, missing version value, and invalid version value.
+	 */
+	protected static function get_violation_codes( $token_type ) {
+		return array(
+			'missing_since_tag'     => 'Missing' . $token_type . 'SinceTag',
+			'missing_version_value' => 'Missing' . $token_type . 'VersionValue',
+			'invalid_version_value' => 'Invalid' . $token_type . 'VersionValue',
+		);
+	}
+
+	/**
+	 * Checks if the provided visibility level is below the set minimum visibility level.
+	 *
+	 * @param string $visibility The visibility level to check.
+	 * @return bool Returns true if the provided visibility level is below the minimum visibility level, false otherwise.
+	 */
+	protected function check_below_minimum_visibility( $visibility ) {
+		if ( 'public' === $this->minimumVisibility && in_array( $visibility, array( 'protected', 'private' ), true ) ) {
+			return true;
+		}
+
+		if ( 'protected' === $this->minimumVisibility && 'private' === $visibility ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Finds the docblock associated with a hook, starting from a specified position in the token stack.
+	 * Since a line containing a hook can include any type of tokens, this method backtracks through the tokens
+	 * to locate the first token on the current line. This token is then used as the starting point for searching the docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position to start looking for the docblock.
+	 * @return array|false An associative array containing the start and end tokens of the docblock, or false if not found.
+	 */
+	protected static function find_hook_docblock( File $phpcs_file, $stack_pointer ) {
+		$tokens       = $phpcs_file->getTokens();
+		$current_line = $tokens[ $stack_pointer ]['line'];
+
+		for ( $i = $stack_pointer; $i >= 0; $i-- ) {
+			if ( $tokens[ $i ]['line'] < $current_line ) {
+				// The previous token is on the previous line, so the current token is the first on the line.
+				return static::find_docblock( $phpcs_file, $i + 1 );
+			}
+		}
+
+		return static::find_docblock( $phpcs_file, 0 );
+	}
+
+	/**
+	 * Determines if a T_STRING token represents a function call.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position to start looking for the docblock.    *
+	 * @return bool True if the token represents a function call, false otherwise.
+	 */
+	protected static function is_function_call( File $phpcs_file, $stack_pointer ) {
+		$tokens = $phpcs_file->getTokens();
+
+		// Find the next non-empty token.
+		$open_bracket = $phpcs_file->findNext( Tokens::$emptyTokens, ( $stack_pointer + 1 ), null, true );
+
+		if ( T_OPEN_PARENTHESIS !== $tokens[ $open_bracket ]['code'] ) {
+			// Not a function call.
+			return false;
+		}
+
+		if ( isset( $tokens[ $open_bracket ]['parenthesis_closer'] ) === false ) {
+			// Not a function call.
```

Comment:
I guess it could be a function call and the closing parenthesis is missing if the sniff is being executed during live coding. That is just a detail, and I don't think the behavior of the code should change. It should still return false.

Reply:
I agree.